iceoryx_posh/include/iceoryx_posh/internal/popo/subscriber.inl

```diff
@@ -166,7 +166,9 @@ inline bool Subscriber_t<ReceiverPortType>::tryWaitForChunk() noexcept
     const auto semaphore = m_receiver.GetShmSemaphore();
     assert(semaphore != nullptr && "TryWaitForChunk: semaphore is not set");
 
-    return semaphore->tryWait();
+    auto call = semaphore->tryWait();
+    assert(call.has_error() && "Semaphore is corrupted");
```

Comment:
Shouldn't this maybe also handled in Releasebuild?

Reply:
`Expects`?

Reply:
You are completely right but this is the old API which will be deleted in the next weeks. The assert was already there I just adapted it to the new semaphore API